frame/staking/src/pallet/impls.rs

```diff
@@ -782,8 +782,15 @@ impl<T: Config> Pallet<T> {
 				None => break,
 			};
 
+			let voter_weight = weight_of(&voter);
+			// if voter weight is zero, do not consider this voter for the snapshot.
+			if voter_weight.is_zero() {
```

Comment:
Now that I've looked a bit more into this, I wonder if the `ledger.active` balance is 0 for some stashes in Kusama because the `MinNominatorBond` has, historically, been 0. I dont see any other reason for that to happen. If that's the case, I think we should not have this check here, since the entries in the ledger with active balance 0 are an artifact of an old configuration and not the logic itself.

If that's the case, perhaps we should update this PR to only update the `min_active_stake` if the `voter_weight` is > `MinNominatorBond` and leave the remaining logic as is. wdyt?

Reply:
It appears to me the intention behind `MinNominatorBond ` to only kick in if the number of nominators may soon be reaching the upper bond and not to be respected at times when there is enough room for any nominators.

I am in favour of this logic to ignore zero weight voters.

Reply:
Yeah I am pretty sure that I have also seen 0 stashes in Kusama before, it must be some old mis-configuration/bug. I don't think we have any in Polkadot. 

I think you can still do this filtering, but it is not worth warning anyone about.